File: Python3/Hard/WrongAnswer/count-of-smaller-numbers-after-self_3.py
class Solution:
    def countSmaller(self, nums: List[int]) -> List[int]:
        
        n = len(nums)
        if n == 0: return []
        if n == 1: return [0]
        mid = n // 2
        left = self.countSmaller(nums[:mid])
        right = self.countSmaller(nums[mid:])
        res = [0] * n
        i = j = 0
        while i < len(left) and j < len(right):
            if nums[mid + i] < nums[j]:
                res[i + j] += left[i] + 1
                i += 1
            else:
                res[i + j] += right[j]
                j += 1
        while i < len(left):
            res[i + j] += left[i]
            i += 1
        while j < len(right):
            res[i + j] += right[j]
            j += 1
        return res

File: Rust/Hard/WrongAnswer/count-of-smaller-numbers-after-self_3.rs
impl Solution {
    pub fn count_smaller(nums: Vec<i32>) -> Vec<i32> {
        
        let mut counts = vec![0; nums.len()];
        let mut sorted = vec![];
        for (i, &n) in nums.iter().rev().enumerate() {
            let pos = sorted.binary_search(&n).unwrap_or_else(|x| x);
            counts[nums.len() - 1 - i] = pos as i32;
            sorted.insert(pos, n);
        }
        counts
    }
}